lib/file_validators/validators/file_content_type_validator.rb

```diff
@@ -46,35 +49,13 @@ def parse_values(value)
         Array.wrap(value).reject(&:blank?)
       end
 
-      def get_file_path(value)
-        if value.try(:path)
-          value.path
-        else
-          raise ArgumentError, 'value must return a file path in order to validate file content type'
-        end
-      end
-
-      def get_file_name(value)
-        if value.try(:original_filename)
-          value.original_filename
-        else
-          File.basename(get_file_path(value))
+      def get_content_type(value, tool)
+        if value.is_a?(Hash)
+          value = OpenStruct.new(value)
+          return value.content_type
```

Comment:
This `value.content_type` is provided by browser's form. We cannot assume that the provided content types are secure. We use it only when modes (:strict, :relaxed) are not set.

Reply:
I've added some tests...

Reply:
I've preserved current behaviour. I'm not sure for what this is used for, maybe for files provided via JSON String?

Reply:
Yes. It can be either JSON string or Hash.

The current behaviour would be that if any of the modes is set then we call the mime type analyzer. Otherwise, we return the provided content. Better if you keep the previous code in the `else` block:
```ruby
value = OpenStruct.new(value) if value.is_a?(Hash)
value.content_type
```
----

The snippet you wrote should have failed. I think I didn't add any spec regarding this behaviour. It would be cool if you add a failing spec then change the code as suggested. Thanks.

Reply:
I'm wondering if we should raise exception if tool is used in combinations with `Hash`?

```ruby
      def get_content_type(value, tool)
        if tool.present?
          unless value.respond_to?(:read)
            raise FileValidators::Error, "value must be valid io in order to validate file content type"
          end

          return FileValidators::MimeTypeAnalyzer.new(tool).call(value)
        end

        value = OpenStruct.new(value) if value.is_a?(Hash)
        value.content_type
      end
```